app/services/proforma_service/convert_exercise_to_task.rb

```diff
@@ -168,8 +190,11 @@ def task_files
     end
 
     def task_file(file)
+      file.update(xml_id_path: xml_id_from_file(file)) if file.xml_id_path.blank?
+
+      xml_id = xml_id_from_file(file).last
```

Comment:
This seems weird. Why not encapsulate the `update` within the `xml_id_from_file` (so that it always returns the XML ID (even if it needs to be created first)?

Reply:
good idea